073-decouple-ca-management-from-end-entity-certificate-management.md

```diff
@@ -0,0 +1,339 @@
+The second Kubernetes Secret, called <CLUSTER_NAME>-cluster-ca-trusted-certs (referred to from here onwards as trusted-certs Secret) is used to store all Cluster CA certificates that are currently trusted by components.
+The trusted-certs Secret is mounted onto Kubernetes Pods and used for their TrustStore.
+These new Kubernetes Secrets are separate from the ones that are used by Strimzi to store the latest CA certificate and key (or by the user when they are managing their own CA certificates).
+The cluster operator will update these new Kubernetes Secrets when it renews the cluster CA certificate (or observes that the user has updated them).
+
+Introducing these new Kubernetes Secrets brings a couple of benefits:
+* The existing Kubernetes Secrets used by the user or Strimzi to store the latest CA certificate are no longer directly mounted onto Kubernetes Pods.
+  This allows Strimzi to more carefully manage when the latest CA certificate is trusted by components.
+  If in future components are able to dynamically update their TrustStores, updating the Kubernetes Secret storing the CA certificate doesn't automatically update the CA certificate being trusted.
+  This also means when integrating with a certificate management solution Strimzi doesn't need to reuse these Kubernetes Secrets, it can fetch certificates programmatically for example.
+* It will be easier to reason about which CA certificates are in use/trusted, both as an end user when viewing the cluster, but also as a developer when reviewing the code.
+  This is because currently the state is inferred by observing various Kubernetes annotations and Java class fields.
+
+These benefits will also make it easier to evolve the management of CA certificates in future (see [Future changes](#future-changes)).
+
+Non-goals of the proposal:
+* This proposal does not change the management of Clients CA.
+  Users can provide custom listener certificates, so allowing integration with a certificate management solution for the Clients CA doesn't bring as many benefits as it does for the Cluster CA.
+* This proposal does not introduce integration with a certificate management solution.
+  The aim of this proposal is to make it easier to allow this kind of integration in future, but assumes that the Cluster CA is still managed by Strimzi and (other than during renewal) there is a single Cluster CA certificate for the cluster.
+
+### New trust-state Kubernetes Secret
+
+A new Kubernetes Secret is introduced which is updated by the cluster operator in its role of managing the cluster (specifically which issued certificates are stored where and rolling pods to use/trust those certificates) to reflect whether the individual CA certificates are trusted and/or in use in the cluster.
+
+```yaml
+kind: Secret
+metdata:
+  name: <CLUSTER_NAME>-cluster-ca-trust-state # e.g. foo-cluster-ca-trust-state
+type: strimzi.io/trust
+data:
+  ${fingerprint}.${state}: <PEM encoded CA certificate>
```

Comment:
Do we still need to have the PEM encoded certificate here? Maybe ${fingerprint}: ${state} is enough now.